File: src/main/java/practice/dto/Customer.java
package practice.dto;

import javafx.beans.property.SimpleStringProperty;
import lombok.Data;

@Data
public class Customer {
    private final SimpleStringProperty firstName = new SimpleStringProperty("");
    private final SimpleStringProperty lastName = new SimpleStringProperty("");

    public Customer() {
        this("", "");
    }

    public Customer(String firstName, String lastName) {
        setFirstName(firstName);
        setLastName(lastName);
    }

    public String getFirstName() {
        return firstName.get();
    }

    public void setFirstName(String fName) {
        firstName.set(fName);
    }

    public String getLastName() {
        return lastName.get();
    }

    public void setLastName(String fName) {
        lastName.set(fName);
    }
}

File: src/main/java/practice/dto/Customer.kt
package practice.dto

import javafx.beans.property.SimpleStringProperty

class Customer @JvmOverloads constructor(firstName: String? = "", lastName: String? = "") {
    private val firstName = SimpleStringProperty("")
    private val lastName = SimpleStringProperty("")

    init {
        setFirstName(firstName)
        setLastName(lastName)
    }

    fun getFirstName(): String {
        return firstName.get()
    }

    fun setFirstName(fName: String?) {
        firstName.set(fName)
    }

    fun getLastName(): String {
        return lastName.get()
    }

    fun setLastName(fName: String?) {
        lastName.set(fName)
    }
}